src/frontend/main.go

```diff
@@ -225,7 +280,9 @@ func mustConnGRPC(ctx context.Context, conn **grpc.ClientConn, addr string) {
 	*conn, err = grpc.DialContext(ctx, addr,
 		grpc.WithInsecure(),
 		grpc.WithTimeout(time.Second*3),
-		grpc.WithStatsHandler(&ocgrpc.ClientHandler{}))
+		grpc.WithStatsHandler(&ocgrpc.ClientHandler{}),
+		grpc.WithUnaryInterceptor(grpctrace.UnaryClientInterceptor(global.TraceProvider().Tracer("frontend"))),
+		grpc.WithStreamInterceptor(grpctrace.StreamClientInterceptor(global.TraceProvider().Tracer("frontend"))))
```

Comment:
I see you use interceptor. There should be a way to detect the name of the grpc action that is being called, so you shouldn't need to call 
```
trace.SpanFromContext(r.Context()).SetAttributes(kv.String("name", "home"))
```
and similar in each method

Reply:
removed and added it to `ServeHTTP` with `r.URL.Path` instead